crates/ruff_server/src/format.rs

```diff
@@ -12,8 +12,16 @@ pub(crate) fn format(
     formatter_settings: &FormatterSettings,
 ) -> crate::Result<String> {
     let format_options = formatter_settings.to_format_options(source_type, document.contents());
-    let formatted = format_module_source(document.contents(), format_options)?;
-    Ok(formatted.into_code())
+    match format_module_source(document.contents(), format_options) {
+        Ok(formatted) => Ok(formatted.into_code()),
+        // Special case - syntax/parse errors should be handled here instead of
+        // being propagated to become visible server errors.
```

Comment:
Nit: reword for clarity?

```suggestion
        // Special case - syntax/parse errors are be handled here instead of
        // being propagated as visible server errors.
```